tests/test_isrTask.py

```diff
@@ -0,0 +1,529 @@
+        self.dataRef = isrMock.DataRefMock()
+        self.camera = isrMock.IsrMock().getCamera()
+
+        self.inputExp = isrMock.TrimmedRawMock().run()
+        self.amp = self.inputExp.getDetector()[0]
+        self.mi = self.inputExp.getMaskedImage()
+
+    def validateIsrData(self, results):
+        """results should be a struct with components that are
+        not None if included in the configuration file.
+        """
+        self.assertIsInstance(results, Struct)
+        if self.config.doBias is True:
+            self.assertIsNotNone(results.bias)
+        if self.config.doDark is True:
+            self.assertIsNotNone(results.dark)
+        if self.config.doFlat is True:
+            self.assertIsNotNone(results.flat)
+        if self.config.doFringe is True:
+            self.assertIsNotNone(results.fringes)
+        if self.config.doDefect is True:
+            self.assertIsNotNone(results.defects)
+        if self.config.doBrighterFatter is True:
+            self.assertIsNotNone(results.bfKernel)
+        if self.config.doAttachTransmissionCurve is True:
+            self.assertIsNotNone(results.opticsTransmission)
+            self.assertIsNotNone(results.filterTransmission)
+            self.assertIsNotNone(results.sensorTransmission)
+            self.assertIsNotNone(results.atmosphereTransmission)
+
+    def test_readIsrData_noTrans(self):
+        """Test that all necessary calibration frames are retrieved.
+        """
+        self.config.doAttachTransmissionCurve = False
+        self.task = IsrTask(config=self.config)
+        results = self.task.readIsrData(self.dataRef, self.inputExp)
+        self.validateIsrData(results)
+
+    def test_readIsrData_withTrans(self):
+        """Test that all necessary calibration frames are retrieved.
+        """
+        self.config.doAttachTransmissionCurve = True
+        self.task = IsrTask(config=self.config)
+        results = self.task.readIsrData(self.dataRef, self.inputExp)
+        self.validateIsrData(results)
+
+    def test_ensureExposure(self):
+        """Test that an exposure has a usable instance class.
+        """
+        self.assertIsInstance(self.task.ensureExposure(self.inputExp, self.camera, 0),
+                              afwImage.Exposure)
+
+    def test_convertItoF(self):
+        """Test conversion from integer to floating point pixels.
+        """
+        self.assertIsInstance(self.task.convertIntToFloat(self.inputExp).getImage()[1, 1],
+                              float)
+
+    def test_updateVariance(self):
+        """Expect The variance image should have a larger median value after
+        this operation.
+        """
+        statBefore = computeImageMedianAndStd(self.inputExp.variance[self.amp.getBBox()])
+        self.task.updateVariance(self.inputExp, self.amp)
+        statAfter = computeImageMedianAndStd(self.inputExp.variance[self.amp.getBBox()])
+        self.assertGreater(statAfter[0], statBefore[0])
```

Comment:
I would make any test that is doing `assertGreater` or `assertLess` use the empirical values instead. That way we at least know if some code causes a change in the algorithm. So here, make the test:

```
self.assertFloatsEqual(statBefore == (123, 0.432))
self.assertFloatsEqual(statBefore == (567, 0.432))
```

or whatever the current output is.

Reply:
I've included both, because even with the comment indicating which way this should go, I would like to have it explicitly tested for (if for no other reason than to prevent me from changing the values blindly in the future).